packages/starlight/utils/route-data.ts

```diff
@@ -105,10 +103,8 @@ function getEditUrl({ entry, id, isFallback }: PageProps): URL | undefined {
 		// If a URL was provided in frontmatter, use that.
 		url = editUrl;
 	} else if (config.editLink.baseUrl) {
-		const srcPath = project.srcDir.replace(project.root, '');
-		const filePath = isFallback ? localizedId(id, config.defaultLocale.locale) : id;
 		// If a base URL was added in Starlight config, synthesize the edit URL from it.
-		url = ensureTrailingSlash(config.editLink.baseUrl) + srcPath + 'content/docs/' + filePath;
+		url = ensureTrailingSlash(config.editLink.baseUrl) + entry.filePath;
```

Comment:
What happens here with legacy collections? Is `filePath` still available?

**Edit:** Ah, I see now in `normalizeCollectionEntry()` you ensure it is present for those entries too 👍